### DISCUSSION TITLE
How to select a particular formatter among many?

### DISCUSSION THREAD

**Discussion by V4G4X** [neovim/neovim#29747] [Q&A]
Suppose I have a bunch of formatters available.
1. yamlls (General YAML, From Mason)
2. spectral (For OpenAPI, From Mason)
3. cfn-lint (For AWS CFN, installed by Mason, enabled by null-ls using `null_ls.builtins.diagnostics.cfn_lint`)
4. Prettier (General case)

How do I?
1. Set a default formatter for a file type.
    Is it possible to override this at a workspace level? 
    So that `spectral` is the default for one project, and `prettier` is the default for the rest?
2. Run a one-off format command with a particular formatter?
    So `cfn-lint` may be set as the default formatter, but just this once I want to format the file with `prettier`.

I think I may be asking for a lot but this seems like a common use case, especially if you come from VS Code. 
Please let me know what can be done!

**Comment by Hamza12700** [comment]
You can use `ftplugin` directory to override the default settings on that filetype.
To learn more about it `:help ftplugin`
